#goal
bool start
---
#result
string end_condition
---
#feedback
bool nothing